components/shared_code/tests/shared/model/test_metric.py

```diff
@@ -25,91 +25,91 @@ def create_metric(self, **kwargs) -> Metric:
+    ) -> dict[str, None | str | list | dict]:
+        """Return an expected summary."""
+        if measurement_timestamp:
+            latest_measurement = {
+                "count": {"value": 1, "start": measurement_timestamp},
+                "end": measurement_timestamp,
+                "sources": [],
+                "start": measurement_timestamp,
+                "status": status,
+            }
+            recent_measurements = [
+                {
+                    "count": {"status": status, "value": 1},
+                    "end": measurement_timestamp,
+                    "start": measurement_timestamp,
+                },
+            ]
+        else:
+            latest_measurement = None
+            recent_measurements = []
+        summary = {
+            "type": "fixture_metric_type",
+            "scale": "count",
+            "status": status if status else None,
+            "status_start": None,
+            "latest_measurement": latest_measurement,
+            "recent_measurements": recent_measurements,
+            "sources": {},
+            **kwargs,
+        }
+        if measurement_timestamp:
+            summary["issue_status"] = []
+        if outdated:
+            summary["latest_measurement"]["outdated"] = True
+        return summary
+
     def test_summarize_empty_metric(self):
         """Test that a minimal metric returns a summary."""
         summary = self.create_metric().summarize([])
-        self.assertDictEqual(
-            summary,
-            {
-                "type": "fixture_metric_type",
-                "scale": "count",
-                "status": None,
-                "status_start": None,
-                "latest_measurement": None,
-                "recent_measurements": [],
-                "sources": {},
-            },
+        self.assertDictEqual(summary, self.create_expected_summary())
+
+    def test_summarize_metric_with_a_measurement(self):
+        """Test that the metric summary includes the measurement."""
+        metric = self.create_metric()
+        measurement_timestamp = iso_timestamp()
+        measurement = Measurement(
+            metric,
+            count={"value": 1, "start": measurement_timestamp},
+            status="target_met",
         )
+        summary = metric.summarize([measurement])
+        self.assertDictEqual(summary, self.create_expected_summary("target_met", measurement_timestamp))
 
-    def test_summarize_measurements(self):
-        """Test that a minimal metric returns a summary."""
+    def test_summarize_metric_with_an_outdated_measurement(self):
+        """Test that the metric is marked outdated when the source parameters have changed."""
         metric = self.create_metric()
         measurement_timestamp = iso_timestamp()
         measurement = Measurement(
             metric,
             count={"value": 1, "start": measurement_timestamp},
             status="target_met",
+            source_parameter_hash="old hash",
```

Comment:
That's not actually changing the source parameters, but a sneaky shortcut :)))

Reply:
Yup :-)